test/unit/package_test.rb

```diff
@@ -1,68 +1,403 @@
-      assert_equal 120 * GRAMS_IN_AN_OUNCE, wii.send(sym)
-    end
-    [:kilograms, :kgs, :kg].each do |sym|
-      assert_equal 120 * GRAMS_IN_AN_OUNCE / 1000, wii.send(sym)
-    end
-    assert_equal 675.0, wii.inches(:volume)
-    assert_equal 675.0, wii.inches(:box_volume)
+    @weight = 100
+    @dimensions = [5, 6, 7]
+    @value = 1299
+    @currency = "USD"
+    @cylinder = false
+    @tube = false
+    @gift = false
+    @oversized = false
+    @unpackaged = false
+    @dim_units = :metric
+    @units = :metric
+    @weight_units = :metric
+    @options = {
+      value: @value,
+      currency:  @currency,
+      cylinder: @cylinder,
+      tube: @tube,
+      gift: @gift,
+      oversized: @oversized,
+      unpackaged: @unpackaged,
+      dim_units: @dim_units,
+      units: @units,
+      weight_units: @weight_units,
+    }
+
+    @package = Package.new(@weight, @dimensions, @options)
+    @imperial_package = Package.new(@weight, @dimensions, @options.merge(units: :imperial, dim_units: :imperial, weight_units: :imperial))
 
-    assert_equal 'GBP', wii.currency
-    assert_equal 26999, wii.value
+    @mass = ::Quantified::Mass.new(@weight, :grams)
   end
 
   def test_package_from_mass
-    pkg = Package.new(Quantified::Mass.new(10, :pounds), [])
-    assert_equal 10, pkg.weight
+    package = Package.new(Quantified::Mass.new(10, :pounds), [])
+    assert_equal 10, package.weight
+  end
+
+  def test_initialize_defaults
+    assert_equal @value, @package.value
+    assert_equal @currency, @package.currency
+    assert_equal @cylinder, @package.cylinder?
+    assert_equal @tube, @package.tube?
+    assert_equal @oversized, @package.oversized?
+    assert_equal @unpackaged, @package.unpackaged?
+    assert_equal @gift, @package.gift?
+  end
+
+  def test_currency_cents
+    @package = Package.new(@weight, @dimensions, value: money)
+    assert_equal @currency, @package.currency
+    assert_equal @value, @package.value
+  end
+
+  def test_initialize_sorted_dimensions
+    @package = Package.new(@weight, [9, 8, 7], @options)
+
+    assert_equal [7, 8, 9], @package.centimetres
+  end
+
+  def test_initialize_blank_dimensions
+    @package = Package.new(@weight, [], @options)
+
+    assert_equal [0, 0, 0], @package.centimetres
+  end
+
+  def test_initialize_increases_dimension_size_to_three
+    @package = Package.new(@weight, [2], @options)
+
+    assert_equal [2, 2, 2], @package.centimetres
+  end
+
+  def test_initialize_default_units
+    assert_equal @dimensions, @package.centimetres
+    assert_equal @weight, @package.grams
+  end
+
+  def test_initialize_units
+    assert_equal @dimensions, @imperial_package.inches
+  end
+
+  def test_initialize_weight_units
+    @package = Package.new(@weight, @dimensions, @options.merge(weight_units: :imperial))
+
+    assert_equal @weight, @package.ounces
+  end
+
+  def test_unpackaged
+    assert Package.new(@weight, @dimensions, unpackaged: true).unpackaged?
```

Comment:
Always avoid bare `assert` / `refute` (bare meaning without an error message) since the failed assertion message is useless. `assert_predicate Package.new(...), :unpackaged?` for this, and anywhere else here.